qualtran/bloqs/arithmetic/comparison.py

```diff
@@ -465,50 +467,172 @@ def _has_unitary_(self):
 class GreaterThan(Bloq):
     r"""Compare two integers.
 
-    Implements $U|a\rangle|b\rangle|0\rangle \rightarrow
-    |a\rangle|b\rangle|a > b\rangle$ using $8n T$  gates.
+    Implements |a>|b>|t> => |a>|b>|t ⨁ (b > a)> using $4n$ T gates.
```

Comment:
This flips the gate. This gate is supposed to do `a > b` not `b > a` as is written in the deleted docstring

$$U|a\rangle|b\rangle|0\rangle \rightarrow
    |a\rangle|b\rangle|a > b\rangle$$

The change is why it is failing the tests. if that is intended then I think you should leave the `GreaterThan` bloq as is and implement a `LessThan` bloq